code/modules/mob/living/bloodcrawl.dm

```diff
@@ -37,58 +40,111 @@
+	if(pulledby)
+		pulledby.stop_pulling()
+	src.loc = holder
+	src.holder = holder
+
+	// if we're not pulling anyone, or we can't eat anyone
+	if(!src.pulling || src.bloodcrawl != BLOODCRAWL_EAT)
+		return
+
+	// if the thing we're pulling isn't alive
+	if (!(istype(src.pulling, /mob/living)))
+		return
+
+	var/mob/living/victim = src.pulling
+	var/kidnapped = FALSE
+
+	if(victim.stat == CONSCIOUS)
+		src.visible_message("<span class='warning'>[victim] kicks free of the blood pool just before entering it!</span>", null, "<span class='notice'>You hear splashing and struggling.</span>")
+	else if(victim.reagents && victim.reagents.has_reagent("demonsblood"))
+		visible_message("<span class='warning'>Something prevents [victim] from entering the pool!</span>", "<span class='warning'>A strange force is blocking [victim] from entering!</span>", "<span class='notice'>You hear a splash and a thud.</span>")
+	else
+		victim.loc = src
+		victim.emote("scream")
+		src.visible_message("<span class='warning'><b>[src] drags [victim] into the pool of blood!</b></span>", null, "<span class='notice'>You hear a splash.</span>")
+		kidnapped = TRUE
+
+	if(kidnapped)
+		var/success = bloodcrawl_consume(B, victim)
+		if(!success)
+			src << "<span class='danger'>You happily devour... nothing? Your meal vanished at some point!</span>"
 	return 1
 
+/mob/living/proc/bloodcrawl_consume(obj/effect/decal/cleanable/B, mob/living/victim)
+	src << "<span class='danger'>You begin to feast on [victim]. You can not move while you are doing this.</span>"
+
+	var/sound
+	if(istype(src, /mob/living/simple_animal/slaughter))
+		var/mob/living/simple_animal/slaughter/SD = src
+		sound = SD.feast_sound
+	else
+		sound = 'sound/magic/Demon_consume.ogg'
+
+	for(var/i in 1 to 3)
+		playsound(get_turf(src),sound, 100, 1)
+		sleep(30)
+
+	if(!victim)
+		return FALSE
+
+	if(victim.reagents && victim.reagents.has_reagent("devilskiss"))
+		src << "<span class='warning'><b>AAH! THEIR FLESH! IT BURNS!</b></span>"
+		adjustBruteLoss(25) //I can't use adjustHealth() here because bloodcrawl affects /mob/living and adjustHealth() only affects simple mobs
+		victim.loc = get_turf(B)
+		// This whole thing here is the ONLY reason we keep passing B down
+		// this proc chain, so we know where to spit the person out who's
+		// been drinking. TODO would be a lot better if it just picked the
+		// nearest bloodspot and used that, then it'd be a lot cleaner
+		// from a code perspective
+		victim.visible_message("<span class='warning'>[B] violently expels [victim]!</span>")
+		return TRUE
+
+	src << "<span class='danger'>You devour [victim]. Your health is fully restored.</span>"
+	src.adjustBruteLoss(-1000)
+	src.adjustFireLoss(-1000)
+	src.adjustOxyLoss(-1000)
+	src.adjustToxLoss(-1000)
+
+	// No defib possible after laughter
+	victim.adjustBruteLoss(1000)
+	victim.death()
+
+	var/del_corpse = TRUE
+	var/mob/living/simple_animal/slaughter/SD
+	if(istype(src, /mob/living/simple_animal/slaughter))
```

Comment:
That check isn't there because of slaughter demons vs laughter demons. It's there because non-slaughter demons could gain BLOODCRAWL_EAT. And they won't have a consumed_mobs var.